homeassistant/components/playstation_network/media_player.py

```diff
@@ -0,0 +1,133 @@
+
+from dataclasses import dataclass
+import logging
+
+from homeassistant.components.media_player import (
+    MediaPlayerDeviceClass,
+    MediaPlayerEntity,
+    MediaPlayerEntityDescription,
+    MediaPlayerState,
+    MediaType,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import PlaystationNetworkCoordinator
+from .coordinator import PlaystationNetworkData
+from .entity import PlaystationNetworkEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class PSNdata:
+    """PSN dataclass."""
+
+    account = {"id": "", "handle": ""}
+    presence = {"availability": "", "lastAvailableDate": ""}
+    platform = {"status": "", "platform": ""}
+    title = {"name": "", "format": "", "imageURL": None, "playing": False}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Media Player Entity Setup."""
+    coordinator: PlaystationNetworkCoordinator = config_entry.runtime_data
+
+    if coordinator.data.platform.get("platform") is None:
+        username = coordinator.data.username
+        _LOGGER.warning(
+            "No console found associated with account: %s. -- Skipping creation of media player",
+            username,
+        )
+        return
+
+    async_add_entities([MediaPlayer(coordinator)])
+
+
+class MediaPlayer(PlaystationNetworkEntity, MediaPlayerEntity):
+    """Media player entity representing currently playing game."""
+
+    entity_description = MediaPlayerEntityDescription(
+        key="console",
+        translation_key="console",
+        device_class=MediaPlayerDeviceClass.TV,
+    )
+
+    def __init__(self, coordinator: PlaystationNetworkCoordinator) -> None:
+        """Initialize PSN MediaPlayer."""
+        super().__init__(coordinator)
+        self.psn: PlaystationNetworkData = self.coordinator.data
+        _attr_translation_key = "spotify"
+
+    @property
+    def media_image_remotely_accessible(self) -> bool:
+        """Is media image remotely accessible getter."""
+        return True
+
+    @property
+    def state(self) -> MediaPlayerState:
+        """Media Player state getter."""
+        match self.psn.platform.get("onlineStatus", ""):
+            case "online":
+                if (
+                    self.psn.available is True
+                    and self.psn.title_metadata.get("npTitleId") is not None
+                ):
+                    return MediaPlayerState.PLAYING
+                return MediaPlayerState.ON
+            case "offline":
+                return MediaPlayerState.OFF
+            case _:
+                return MediaPlayerState.OFF
+
+    @property
+    def unique_id(self) -> str:
+        """Unique ID Getter."""
+        return f"{self.psn.username.lower()}_{self.psn.platform.get('platform', "").lower()}_console"
+
```

Comment:
```suggestion
        self._attr_unique_id = (
            f"{coordinator.config_entry.unique_id}_{self.entity_description.key}"
        )
   
```

preferably add something like this to the init. Also the unique id should never change for an entity. Instead of key you could also use the platform (ps4, ps5) but this should be passed to the entity when it is created in the entry setup

Reply:
What are your thoughts? 
Would we want to create systems all the way back to the PSP? 
Maybe have some arbitrary cutoff? (I wish I had last used date, then you could create it dynamically if it does comeback online)
The data is a little wonky. I have 2 PS3s (3 in the list), I have 1 Vita (2 unique in the list), I have 2 PS4s (1 in the list). We should condense it down to one for each deviceType, right? 

![Screenshot 2024-11-26 at 8 32 45 AM](https://github.com/user-attachments/assets/2aaab29f-353f-47c5-b9c8-a0e2ae4f21dd)

Reply:
Maybe only the systems that actually report the status to PSN? My PS3 doesn't show me as online anymore. Or create the devices when they were first seen online

Reply:
So I went down the road of creating a PS4 and PS5 media player entity. PS6 would automatically be created too. However, I only get the status of the last system that launched a game. Here is a picture of the problem. I'm playing a game on both my ps4 and ps5 at the same time but I only get details for one of them. If I could at least tell if the system was on or not it would be one thing, but the picture is just wrong. 

So I see two options:
1. This is a contrived problem and isn't something to likely happen in practice and can be called out in the documentation.
2. We just create a single media player that is called "Playstation Console" (something generic like that) and display whatever the api returns. 

![CleanShot 2024-12-13 at 12 43 49](https://github.com/user-attachments/assets/b8daf5ca-b830-419d-aaeb-c3fc06999b04)

Reply:
That looks good! We can put that in the documentation under "known limitations". I only know one persone that would play on 2 consoles at the same time, but only on very rare occasions ^^